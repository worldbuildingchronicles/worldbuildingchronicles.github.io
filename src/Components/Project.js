import React, { Component } from 'react';

import ProjectModal from './ProjectModal';

class Project extends Component { // <a> Can't be nested
  constructor() {
    super();
    this.state = {
      showModal: false
    };
    this.handleOpenModal = this.handleOpenModal.bind(this);
    this.handleCloseModal = this.handleCloseModal.bind(this);
  }

  handleOpenModal() {
    this.setState({ showModal: true });
  }

  handleCloseModal() {
    this.setState({ showModal: false });
  }
  render() {
    const { project } = this.props;
    return (
      <div>
        <ProjectModal
          showModal={this.state.showModal}
          handleCloseModal={this.handleCloseModal}
          project={project} />
        <div onClick={this.handleOpenModal} className="project shadow2 grow">
          <div className="ptext">
            <h1>{project.title}</h1>
            <p>{project.desc}</p>
          </div>
          <div className="bullets">
            <ul>
              {project.badges.slice(0, 2).map((badge) =>
                <div key={badge.text}>
                  <li>{badge.text}</li>
                </div>
              )}
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

export default Project;