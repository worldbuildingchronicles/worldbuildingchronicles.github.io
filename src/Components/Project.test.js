import React from 'react';
import ReactDOM from 'react-dom';
import Project from './Project';
import renderer from 'react-test-renderer';

const lw = {
  title: ' 🌤️ LocalWeather.',
  desc: 'Plaintext description of the weather on top of beautiful relevant Unsplash images.',
  preview: 'images/weather.png',
  badges: [
    {
      text: '🔗 View Project',
      link: '//calumptrck.github.io/LocalWeather/'
    },
  ],
  longDesc: "LocalWeather is a a weather homepage which tells you what's relevant, and nothing else.",
  featured: false,
};

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Project key={"Title"} project={lw} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Project key={"Title"} project={lw} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


});