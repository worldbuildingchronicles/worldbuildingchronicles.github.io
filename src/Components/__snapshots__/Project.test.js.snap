// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div>
  <div />
  <div
    className="project shadow2 grow"
    onClick={[Function]}
  >
    <div
      className="ptext"
    >
      <h1>
         🌤️ LocalWeather.
      </h1>
      <p>
        Plaintext description of the weather on top of beautiful relevant Unsplash images.
      </p>
    </div>
    <div
      className="bullets"
    >
      <ul>
        <div>
          <li>
            🔗 View Project
          </li>
        </div>
      </ul>
    </div>
  </div>
</div>
`;
